### PR TITLE
Implement jetstream requests

### PR DESCRIPTION
**Pull request by caspervonb**

### CODE REVIEW

File: async-nats/src/jetstream/context.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+use crate::Client;
+use bytes::Bytes;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde_json;
+
+/// An error description returned from a jetstream request.
+#[derive(Debug, Deserialize)]
+pub struct Error {
+    /// Code
+    pub code: u64,
```
Comment (Jarema) [inline async-nats/src/jetstream/context.rs:10]: this is supossed to be enum in the future?

Comment (caspervonb) [reply async-nats/src/jetstream/context.rs:10]: Probably remain as a number for raw protocol representation. (this is scoped as async_nats::jetstream::context::Error so while its public, you have to dig to get to it).

For one it's bit problematic to encode an enum from an unknown source of unknown value at this low level call site.

For our actual error handling was thinking a bit more along the lines of how std does it, with for example ParseIntError.

```rust
enum CreateStreamErrorKind {
  AlreadyExists,
  MissingName,
  Other,
}

struct CreateStreamError {}
impl CreateStreamError {
  fn kind() -> CreateStreamErrorKind {}
}

impl Context {
  pub async fn create_stream(stream_config: StreamConfig) -> Result<Stream, CreateStreamError> {}
}
```

File: async-nats/src/jetstream/context.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+pub struct Context {
+    client: Client,
+    prefix: String,
+}
+
+impl Context {
+    pub fn new(client: Client) -> Context {
+        Context {
+            client,
+            prefix: "$JS_API".to_string(),
```
Comment (Jarema) [inline async-nats/src/jetstream/context.rs:35]:
```suggestion
            prefix: "$JS.API".to_string(),
```

**Review by Jarema — CHANGES_REQUESTED** [review]
Looks good, needs just a fix in JS api prefix.

**Comment by caspervonb** [comment]
Found an edge case with requests and $JS prefixes, added it as an ignored test.

**Review by Jarema — APPROVED** [review]
LGTM
